In [208]:
import pandas as pd
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import plotly.express as px

In [196]:
data_crime=pd.read_csv("BPD_Part_1_Victim_Based_Crime_Data.csv")

In [197]:
data_crime.head()

CrimeDate CrimeTime CrimeCode           Location          Description  \
0  09/02/2017  23:30:00       3JK    4200 AUDREY AVE  ROBBERY - RESIDENCE   
1  09/02/2017  23:00:00        7A  800 NEWINGTON AVE           AUTO THEFT   
2  09/02/2017  22:53:00        9S      600 RADNOR AV             SHOOTING   
3  09/02/2017  22:50:00        4C     1800 RAMSAY ST         AGG. ASSAULT   
4  09/02/2017  22:31:00        4E       100 LIGHT ST       COMMON ASSAULT   

  Inside/Outside   Weapon   Post  District      Neighborhood  Longitude  \
0              I    KNIFE  913.0  SOUTHERN          Brooklyn  -76.60541   
1              O      NaN  133.0   CENTRAL    Reservoir Hill  -76.63217   
2        Outside  FIREARM  524.0  NORTHERN    Winston-Govans  -76.60697   
3              I    OTHER  934.0  SOUTHERN  Carrollton Ridge  -76.64526   
4              O    HANDS  113.0   CENTRAL     Downtown West  -76.61365   

   Latitude                       Location 1     Premise  Total Incidents  
0  39.22951  (39.2295100000, -76.6054100000)  ROW/TOWNHO                1  
1  39.31360  (39.3136000000, -76.6321700000)      STREET                1  
2  39.34768  (39.3476800000, -76.6069700000)      Street                1  
3  39.28315  (39.2831500000, -76.6452600000)  ROW/TOWNHO                1  
4  39.28756  (39.2875600000, -76.6136500000)      STREET                1

In [198]:
data_crime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 276529 entries, 0 to 276528
Data columns (total 15 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   CrimeDate        276529 non-null  object 
 1   CrimeTime        276529 non-null  object 
 2   CrimeCode        276529 non-null  object 
 3   Location         274322 non-null  object 
 4   Description      276529 non-null  object 
 5   Inside/Outside   266250 non-null  object 
 6   Weapon           95577 non-null   object 
 7   Post             276305 non-null  float64
 8   District         276449 non-null  object 
 9   Neighborhood     273789 non-null  object 
 10  Longitude        274325 non-null  float64
 11  Latitude         274325 non-null  float64
 12  Location 1       274325 non-null  object 
 13  Premise          265772 non-null  object 
 14  Total Incidents  276529 non-null  int64  
dtypes: float64(3), int64(1), object(11)
memory usage: 31.6+ MB


# EDA

In [199]:
year=data_crime["CrimeDate"].str.extract(r'(\d{4})').rename(columns={0:"year"})
month=data_crime["CrimeDate"].str.extract(r'.(\d{2})').rename(columns={0:"month"})
day=data_crime["CrimeDate"].str.extract(r'(\d{2})').rename(columns={0:"day"})

In [200]:
data_crime.drop(columns=["CrimeDate"],inplace=True)

In [201]:
date=pd.concat([year,month,day],axis=1)
data_crime=pd.concat([date,data_crime],axis=1)

# How to see most Crime in 2017 

In [206]:
most_crime_2017=data_crime[data_crime["year"]=="2017"]

In [235]:
most_crime_2017.groupby('year')["Description"].value_counts()

year  Description         
2017  LARCENY                 6478
      COMMON ASSAULT          5884
      BURGLARY                5147
      AGG. ASSAULT            3908
      LARCENY FROM AUTO       3683
      AUTO THEFT              3195
      ROBBERY - STREET        2538
      ROBBERY - COMMERCIAL     666
      ASSAULT BY THREAT        480
      SHOOTING                 466
      ROBBERY - CARJACKING     371
      ROBBERY - RESIDENCE      361
      HOMICIDE                 237
      RAPE                     226
      ARSON                    184
Name: Description, dtype: int64

In [231]:
style_most_crime= pd.DataFrame(most_crime_2017['Description'].value_counts()).rename({"Description":"Total crime"}, axis = 1).rename_axis("Description").head(10)

In [236]:
px.bar(style_most_crime.reset_index(),x="Description",y="Total crime",color="Description")

Most crime in 2017 the answer is 
* Larceny (Total 6478)
* Common Assault (Total 5884)
* AGG. Assault (Total 3908)
* LARCENY FROM AUTO (Total 3683)
* AUTO THEFT (Total 3683)
* AUTO THEFT (Total 3683)